pipeline.py

```diff
@@ -48,19 +59,32 @@ class CorpusManager:
     """
 
     def __init__(self, path_to_raw_txt_data: str):
-        pass
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._scan_dataset()
 
     def _scan_dataset(self):
         """
         Register each dataset entry
         """
-        pass
+        pathlib_path_to_data = Path(self.path_to_raw_txt_data)
+
+        for file_path in pathlib_path_to_data.iterdir():
+            file_name = file_path.name
+
+            if file_name[-8:] == RAW_FILE_PATH_ENDING:
+                match = re.search(r'\d+', file_name)
+                article_id = int(match.group(0))
```

Comment:
to make you code less nesting, you need to change the way you process conditions in a loop. instead of processing large body in case of "yes", make sure you skip iteration if the answer is "no"